src/platform/silabs/rs911x/BLEManagerImpl.cpp

```diff
@@ -105,17 +103,22 @@ void sl_ble_event_handling_task(void)
     //! This semaphore is waiting for wifi module initialization.
     rsi_semaphore_wait(&sl_rs_ble_init_sem, 0);
 
-    // This function initialize BLE and start BLE advertisement.
+    // Initialize BLE and starts BLE Advertisement
     sl_ble_init();
 
     // Application event map
     while (1)
     {
-        //! This semaphore is waiting for next ble event task
-        rsi_semaphore_wait(&sl_ble_event_sem, 0);
-
         // checking for events list
         event_id = rsi_ble_app_get_event();
+
+        if (event_id == -1)
+        {
+            //! This semaphore is waiting for next ble event task
+            rsi_semaphore_wait(&sl_ble_event_sem, 0);
```

Comment:
This line is useless, unless for some reason a waiting time of 0 is equivalent to an infinite wait. That leads to a follow question : Why use the `rsi_semaphore_wait` and not the standard FreeRTOS API ? This should be change for both code clarity and uniformity across one's stack.